Assets/Tests/PlayMode/PlayerDataStorageManagerTests.cs

```diff
@@ -0,0 +1,326 @@
+
+            // The key is the file name; we need to delete all of these files
+            foreach (string localCacheKey in new List<string>(localCache.Keys))
+            {
+                // To know when the DeleteFile operation is complete, we also listen to the file list being updated
+                waiting = true;
+                playerDataStorageManager.DeleteFile(localCacheKey);
+                yield return new WaitUntilDone(10f, () => waiting == false);
+            }
+
+            // Unsubscribe from file list updates before exiting
+            playerDataStorageManager.OnFileListUpdated -= setWaiting;
+        }
+
+        /// <summary>
+        /// Assuming there are no files in storage for the test user,
+        /// the user should be able to successfully query for items, even if there are none to be found.
+        /// The resulting cached data should be empty.
+        /// </summary>
+        [UnityTest]
+        public IEnumerator Query_NoResults()
+        {
+            bool waiting = true;
+            Action setWaiting = () => waiting = false;
+            playerDataStorageManager.OnFileListUpdated += setWaiting;
+            playerDataStorageManager.QueryFileList();
+
+            yield return new WaitUntilDone(10f, () => waiting == false);
+
+            Dictionary<string, string> localCache = playerDataStorageManager.GetLocallyCachedData();
+
+            Assert.NotNull(localCache, "Local cache is null, should be an empty dictionary.");
+            Assert.AreEqual(0, localCache.Keys.Count, "Local cache contains more than zero items, should be empty.");
+        }
+
+        /// <summary>
+        /// This is used as a datasource for <see cref="UploadedFiles_CanBeQueried"/>.
+        /// Determines how many files to make.
+        /// </summary>
+        static int[] ValueSource_NumberOfFilesToMakeFor_UploadedFiles_CanBeQueried = { 1, 3, 10 };
+
+        /// <summary>
+        /// Creates a file, uploads it, and then checks that it exists in the list.
+        /// </summary>
+        /// <param name="numberOfFilesToMake">
+        /// How many files to create.
+        /// This ensures that the manager can handle single file and multi file scenarios appropriately.
+        /// </param>
+        [UnityTest]
+        public IEnumerator UploadedFiles_CanBeQueried([ValueSource(nameof(ValueSource_NumberOfFilesToMakeFor_UploadedFiles_CanBeQueried))] int numberOfFilesToMake)
+        {
+            HashSet<string> randomNames = new HashSet<string>();
+
+            for (int fileIndex = 0; fileIndex < numberOfFilesToMake; fileIndex++)
+            {
+                // Generate a random name until we hit something that isn't already taken
+                // Having the same name would work as an overwrite
+                string randomName = "";
+                do
+                {
+                    randomName = UnityEngine.Random.Range(0, 9999).ToString();
+                } while (randomNames.Contains(randomName));
```

Comment:
I think it would be far simpler to use `Guid::New()` (or whatever the signature is). That way it is clearer, far less likely to collide, and all the file names will be of the same length.

Reply:
Sure, that's really simple and easy to do. GUID collisions should never happen.